packages/app/src/features/NoteEditor/index.tsx

```
@@ -61,6 +72,244 @@ export enum NoteSidebarTabs {
+			for (let attempt = 0; attempt < 3; attempt++) {
+				try {
+					await noteHistory.snapshot(note.id);
+					res();
+					return;
+				} catch (err) {
+					// Retry after delay
+					console.error(err);
+					await new Promise((res) => setTimeout(res, 200));
+				}
+			}
+		}).then(() => {
```

Biome (2.4.16)

[error] Promise executor functions should not be `async`.

(lint/suspicious/noAsyncPromiseExecutor)